tech(api): update isActive column default value and add column variable

meardagoat opened the pull request:
## ✅ Add isActive column to users table

**Resolves #35**

### Changes Made:
- Created migration file: `20250925085355_add-isActive-column-to-users-table.js`
- Added boolean `isActive` column with default value `true`
- Column is set as `notNullable()` (required field)
- Includes proper rollback functionality

### Migration Details:
```javascript
// Up migration - adds the column
await knex.schema.alterTable("users", (table) => {
  table.boolean("isActive").defaultTo(true).notNullable();
});

// Down migration - removes the column  
await knex.schema.alterTable("users", (table) => {
  table.dropColumn("isActive");
});
```

### Database Schema After Migration:
The users table now includes:
- `id` (primary key)
- `firstname`
- `lastname`
- `email`
- `birthday`
- `isActive` ← **NEW COLUMN** (boolean, default: true, not null)
- `created_at`
- `updated_at`

### Testing:
- ✅ Linter passed with no errors
- ✅ Migration syntax validated
- ✅ Ready for database testing

### To Apply Migration:
```bash
cd api
npm run db:migrate
```

theotime2005 reviewed (changes requested):
The migration is globaly correct but not perfect:)

theotime2005 commented:
In the commit title, you must precises the scope, api, web or other.

meardagoat committed:
tech(api): update isActive column default value and add column variable

- Add boolean isActive column with default value false
- Column is notNullable (required field)
- Add COLUMN_NAME variable for better maintainability
- Includes proper rollback functionality
- Resolves #35

edp5-service commented:
:tada: This PR is included in version 1.0.0 :tada:

The release is available on [GitHub release](https://github.com/edp5/kompagnon/releases/tag/v1.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
